mautic/mautic
# ModelFactory for dynamic model dependency requirements

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y - dev
| Related user documentation PR URL | na
| Related developer documentation PR URL | todo
| Issues addressed (#s or URLs) | na
| BC breaks? | n
| Deprecations? | n

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Since the deprecation of MauticFactory, there are scenarios where it is unknown what models are required and thus dependency injection is impossible. For example, where we have stored channel/source (that correlates with models) in tables. Page hits for example make it possible to get the entity's name by doing something like `$factory->getModel($channel)->getEntity($channelId)`. This PR simply inherits that functionality intended to be used only for such a scenario. This is not meant to "save time" compared to injecting independent models as needed (or we've recreated MauticFactory all over again). 

#### Steps to test this PR:
1. Quick means is to manually edit a page_hit entry injecting the columns source/source_id with `email` and an ID of an existing Mautic email ID. 
2. View the timeline of the contact for that page hit - the name of the email should be noted as teh source.
